### DISCUSSION TITLE
SSL for mysql @ localhost does not work as expected.

### DISCUSSION THREAD

**Discussion by genesem** [PyMySQL/mysqlclient#733] [Issue report]
### Describe the bug

In mysqlclient==2.2.4 this code works:

```
 import MySQLdb

try: myconn = MySQLdb.connect(
   host=host, port=3306, user=myuser, passwd=mypasswd, db=mydbname, 
   ssl_mode='DISABLED',
   ssl=False
   )
except MySQLdb.Error as e:
	print(e) 
	exit(1)
```
after upgrade to latest: mysqlclient==2.2.5
This makes an error:
(2026, 'TLS/SSL error: SSL is required, but the server does not support it')

p.s. mariaDB server running at localhost, with no ssl etc..

### Environment

MariaDB  Ver 15.1 Distrib 10.11.7-MariaDB, for Win64 (AMD64)

[mysqld]
require_secure_transport=OFF

### Minimum but complete code to reproduce

```python
 import MySQLdb

try: myconn = MySQLdb.connect(
   host=host, port=3306, user=myuser, passwd=mypasswd, db=mydbname, 
   ssl_mode='DISABLED',
   ssl=False
   )
except MySQLdb.Error as e:
	print(e) 
	exit(1)
...
```

### Schema and initial data required to reproduce.

```sql
-- Write SQL here.
-- e.g. CREATE TABLE ...
```

**Comment by methane** [comment]
duplicate of #729.
